pkg/reconciler/internal/paramvalidation/paramvalidation.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package paramvalidation
```

Comment:
wondering why creating this new structure `pkg/reconciler/internal/...`? what does this `internal` stand for?

Reply:
this definitely needs unit tests, we might be able to leverage existing tests since the functions are moved out from other packages. 🤔

Reply:
@pritidesai FYI https://go.dev/doc/go1.4#internalpackages

Reply:
+1 to Priti's comment about tests; our code standards state that all exported functions must be tested.

Reply:
sure! I will add tests

Reply:
added

Reply:
> @pritidesai FYI https://go.dev/doc/go1.4#internalpackages

thanks @lbernick for the link, is this a right path for parameter validation `paramvalidation` under `reconciler`?

Reply:
Also, instead of creating a `utility` kind of functions, can we create a member functions to the structs we have defined e.g. `[]Param`?

Reply:
I think the path is OK--moving something out of an internal package is less disruptive than moving something into it, so we can always move it out later. I'd probably use "params" as a package name rather than "paramvalidation". Having it under reconciler is also probably OK since it's validating the values received at runtime but I could see the argument for having it under apis.